msu/systems/serialization/serialization_mod_addon.nut

```diff
@@ -5,4 +5,24 @@
 		local savedVersion = _metaData.getString(this.Mod.getID() + "Version");
 		return savedVersion != "" && ::MSU.SemVer.compareVersionWithOperator(savedVersion, ">=", _version);
 	}
+
+	function flagSerialize( _id, _object, _flags = null, _clear = true )
+	{
+		::MSU.System.Serialization.flagSerialize(this.Mod, _id, _object, _flags, _clear);
+	}
+
+	function flagDeserialize( _id, _object = null, _flags = null, _clear = true )
+	{
+		return ::MSU.System.Serialization.flagDeserialize(this.Mod, _id, _object, _flags, _clear);
+	}
+
+	function flagSerializeBBObject( _id, _bbObject, _flags = null, _clear = true )
```

Comment:
There should be a way to not have to implement these separate BB object functions. Basically the only difference these objects have is that they have their own `onSerialize` function. So all you should do is check `if (::MSU.isIn("onSerialize", _object)` and if yes then do the following two things:
1. Write a string saying "BBObject"
2. Write a string with the filename of the BBObject.
3. Run that function.

If `onSerialize` is not found, then write a string saying "NormalObject" or whatever and then do the normal serialization.

During serialization you always read the string first. And if it says BBObject then you read the next string to get file, then you instantiate it and run its `onDeserialize` function.

Reply:
after some discussion, this should be implemented with merging the two functions, and checking for the presence of release_hook and serialize/deserialize functions to determine whether the BB object serialization should be used.